src/compiler/moduleNameResolver.ts

```diff
@@ -665,16 +665,23 @@ namespace ts {
                 }
             }
 
-            switch (moduleResolution) {
-                case ModuleResolutionKind.NodeJs:
-                    result = nodeModuleNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
-                    break;
-                case ModuleResolutionKind.Classic:
-                    result = classicNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
-                    break;
-                default:
-                    return Debug.fail(`Unexpected moduleResolution: ${moduleResolution}`);
+            try {
+                if (etwLogger) etwLogger.logStartResolveModule(moduleName /* , containingFile, ModuleResolutionKind[moduleResolution]*/);
+                switch (moduleResolution) {
+                    case ModuleResolutionKind.NodeJs:
+                        result = nodeModuleNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
+                        break;
+                    case ModuleResolutionKind.Classic:
+                        result = classicNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
+                        break;
+                    default:
+                        return Debug.fail(`Unexpected moduleResolution: ${moduleResolution}`);
+                }
             }
+            finally {
+                if (etwLogger && result && result.resolvedModule) etwLogger.logInfoEvent(`Module "${moduleName}" resolved to "${result.resolvedModule.resolvedFileName}"`);

```

Comment:
Likewise, maybe in our `PerfLogger` wrapper, we could have a `logInfoEventIf<T extends any[]>(condition: boolean, logText: (...args: T) => string, ...args: T)`, so we can avoid introducing control flow statements for logging into code not related to the logging?

Reply:
I'm not sure I understand this suggestion.

Reply:
Rather than
```ts
if (etwLogger && result && result.resolvedModule) etwLogger.logInfoEvent(`Module "${moduleName}" resolved to "${result.resolvedModule.resolvedFileName}"`)
```
write
```ts
logger.logInfoEvent(result && result.resolvedModule, (moduleName, result) => `Module "${moduleName}" resolved to "${result.resolvedModule.resolvedFileName}"`, moduleName, result);
```
like how our `ts.Debug.assert` signature is setup.

Reply:
How is that preferable?

Reply:
It moves the control flow for the conditional-ness of the logging into the logger, so anyone reading the implementation doesn't need to reason over the effects of the control flow that's only there for the logger. The function-instead-of-string bit is just an optimization to avoid doing expensive string building (like `typeToString`) if the branch isn't hit - it's not strictly needed, and is why `ts.Debug.assert` takes both a string and a function argument.